### PR TITLE
Pixeltable Windows compatibility

### PR DESCRIPTION
**Pull request by aaron-siegel**
Adds Windows CI to Pixeltable, along with a number of fixes to ensure it runs on Windows:
- Add `utf8` as the explicit encoding when opening files as text
- Proper URL parsing for Windows paths (in several places)
- Load `PIL.Image` files immediately after opening them in `data_row`, to ensure the `data_row` does not hold on to a stale file handle indefinitely
- Bump `pgserver` to `0.0.9`
- Disable one test (`test_html_media_url`); the test itself is not portable; this can be addressed later

Also adds a script to automate PyPI releases.

- - -
This change is [Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/121)

### CODE REVIEW

File: pixeltable/tests/test_dataframe.py [hunk 1]
```diff
@@ -183,6 +183,7 @@ def test_select_literal(self, test_tbl: catalog.Table) -> None:
         res = t.select(1.0).where(t.c2 < 10).collect()
         assert res[res.column_names()[0]] == [1.0] * 10
 
+    @pytest.mark.skip('Test is not portable')
```
Comment (mkornacker) [inline pixeltable/tests/test_dataframe.py:187]: Why not?

Leave a todo to fix iti?

File: pixeltable/tests/utils.py [hunk 2]
```diff
@@ -239,9 +239,11 @@ def get_test_video_files() -> List[str]:
     glob_result = glob.glob(f'{tests_dir}/**/test_videos/*', recursive=True)
     return glob_result
 
-def get_image_files() -> List[str]:
+def get_image_files(include_bad_image=False) -> List[str]:
```
Comment (mkornacker) [inline pixeltable/tests/utils.py]: include_bad_image: bool = False

File: scripts/release.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  exit 1
+fi
+
+echo "Project root: $PROJECT_ROOT"
+echo "This will publish version $VERSION. Enter to confirm; Ctrl-C to abort."
+read
+
+cd "$PROJECT_ROOT"
+poetry build
+poetry publish --username __token__ --password "$PYPI_API_KEY"
```
Comment (mkornacker) [inline scripts/release.sh:34]: We should probably also rebuild the docs and publish those.

**Comment by orm011** [comment]
*[`pixeltable/exec/cache_prefetch_node.py` line 86 at r1](https://reviewable.io/reviews/pixeltable/pixeltable/121#-NuyTUms66ckA-VNr1FI:-NuyTUms66ckA-VNr1FJ:b-td34v) ([raw file](https://github.com/pixeltable/pixeltable/blob/71ddafc83eb5bafbb99918196291fa376c678357/pixeltable/exec/cache_prefetch_node.py#L86)):*
> ```Python
>         url = row.file_urls[slot_idx]
>         parsed = urllib.parse.urlparse(url)
>         assert len(parsed.scheme) > 1 and parsed.scheme != 'file'
> ```

this is for stuff like C://? may be good to make a comment.

**Review by orm011 — APPROVED** [review]
:lgtm: (just see comment suggestion)

> Reviewed 12 of 12 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/121#-:-NuyZrfL5R5ERWkKshsF:b-v9npc4)* status: all files reviewed, 4 unresolved discussions (waiting on @aaron-siegel)

**Review by aaron-siegel — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/121)* status: all files reviewed, 4 unresolved discussions (waiting on @mkornacker and @orm011)

___
*[`pixeltable/exec/cache_prefetch_node.py` line 86 at r1](https://reviewable.io/reviews/pixeltable/pixeltable/121#-NuyTUms66ckA-VNr1FI:-Nuy_0vZCQK4XjJQU87_:b-ciy840) ([raw file](https://github.com/pixeltable/pixeltable/blob/71ddafc83eb5bafbb99918196291fa376c678357/pixeltable/exec/cache_prefetch_node.py#L86)):*
<details><summary><i>Previously, orm011 (Oscar Moll) wrote…</i></summary><blockquote>

this is for stuff like C://? may be good to make a comment.
</blockquote></details>

There are lots of comments on this elsewhere in the source but I'll add one here too.
___
*[`pixeltable/tests/test_dataframe.py` line 186 at r1](https://reviewable.io/reviews/pixeltable/pixeltable/121#gh-1556090604:-Nuy_3pl25iH91_qqhVz:b-wlihi1) ([raw file](https://github.com/pixeltable/pixeltable/blob/71ddafc83eb5bafbb99918196291fa376c678357/pixeltable/tests/test_dataframe.py#L186)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

Why not?

Leave a todo to fix iti?
</blockquote></details>

Ok I'll leave a TODO. To be honest I'm not entirely sure what this test is doing, it's kind of weird the way it relies on parsing out _repr_html_. Maybe we can discuss offline.
___
*[`pixeltable/tests/utils.py` line 242 at r1](https://reviewable.io/reviews/pixeltable/pixeltable/121#gh-1556094363:-Nuy_IMk0S7sJ1uGLTT8:b1f1s5o) ([raw file](https://github.com/pixeltable/pixeltable/blob/71ddafc83eb5bafbb99918196291fa376c678357/pixeltable/tests/utils.py#L242)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

include_bad_image: bool = False
</blockquote></details>

Good catch thanks
___
*[`scripts/release.sh` line 34 at r1](https://reviewable.io/reviews/pixeltable/pixeltable/121#gh-1556096038:-Nuy_JUr3cSmhwJqHZbD:b-vzndzk) ([raw file](https://github.com/pixeltable/pixeltable/blob/71ddafc83eb5bafbb99918196291fa376c678357/scripts/release.sh#L34)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

We should probably also rebuild the docs and publish those.
</blockquote></details>

Ok. I also want to automatically create a github release using github CLI. Let me factor this out as a separate problem. I'll put it as a TODO for now

**Comment by orm011** [comment]
*[`pixeltable/tests/test_dataframe.py` line 186 at r1](https://reviewable.io/reviews/pixeltable/pixeltable/121#gh-1556090604:-NuyaTjU7QgsZgxh3CBR:bwgx0wr) ([raw file](https://github.com/pixeltable/pixeltable/blob/71ddafc83eb5bafbb99918196291fa376c678357/pixeltable/tests/test_dataframe.py#L186)):*
<details><summary><i>Previously, aaron-siegel (Aaron Siegel) wrote…</i></summary><blockquote>

Ok I'll leave a TODO. To be honest I'm not entirely sure what this test is doing, it's kind of weird the way it relies on parsing out _repr_html_. Maybe we can discuss offline.
</blockquote></details>

The goal of the test is to check that the repr_html  for media, especially the url set as source is available, to make sure videos/ audio are visible  in the notebook.

**Comment by orm011** [comment]
*[`pixeltable/tests/test_dataframe.py` line 186 at r1](https://reviewable.io/reviews/pixeltable/pixeltable/121#gh-1556090604:-NuyaynRCOin7eifW-Ev:b-4pnxvy) ([raw file](https://github.com/pixeltable/pixeltable/blob/71ddafc83eb5bafbb99918196291fa376c678357/pixeltable/tests/test_dataframe.py#L186)):*
<details><summary><i>Previously, orm011 (Oscar Moll) wrote…</i></summary><blockquote>

The goal of the test is to check that the repr_html  for media, especially the url set as source is available, to make sure videos/ audio are visible  in the notebook.
</blockquote></details>

(NB) the path etc needs to be correct. It used to be broken

**Pull request merged**